### PR TITLE
[Component] [Contentformbuilder] + add ContentFormBuilder + move manage of form from edition plugins

### PR DESCRIPTION
**Pull request by fkroockmann**
I export the element interpreter from the edition plugin to a content.

You can use it like:

```
require('component!contentformbuilder').getConfig('Element/Text').done(function (config) {
      //The config of form
});
```

### CODE REVIEW

File: src/tb/component/contentformbuilder/elements/All.js [hunk 1]
```diff
@@ -38,21 +37,27 @@ define(
...
                     };
 
+                if (object !== undefined) {
+                    this.populateConfig(object, config, dfd);
+                } else {
                     dfd.resolve(config);
-                });
+                }
 
                 return dfd.promise();
```
Comment (ndufreche) [inline src/tb/component/contentformbuilder/elements/All.js]: There is nothing async anymor, you don't need a promise

File: src/tb/component/contentformbuilder/elements/Text.js [hunk 2]
```diff
@@ -38,25 +38,32 @@ define(
...
                     };
 
+                if (object !== undefined) {
+                    this.populateConfig(object, config, dfd);
+                } else {
                     dfd.resolve(config);
-                });
+                }
 
                 return dfd.promise();
```
Comment (ndufreche) [inline src/tb/component/contentformbuilder/elements/Text.js]: Same here, it seems to be not necessary to use the promise

Comment (fkroockmann) [inline src/tb/component/contentformbuilder/elements/Text.js]: It's necessary cause the getData function is async

**Comment by fkroockmann** [comment]
I don't understand why jasmine don't pass, i dont have change the spec :o

File: src/tb/component/contentformbuilder/elements/Text.js (hunk 2 above)
Comment (ndufreche) [inline src/tb/component/contentformbuilder/elements/Text.js]: Ho, ok.

**Comment by mickaelandrieu** [comment]
(relaunched travis)

**Comment by fkroockmann** [comment]
Thanks @mickaelandrieu but i find why

**Comment by mickaelandrieu** [comment]
"Warning: PhantomJS timed out, possibly due to an unfinished async spec.".

**Comment by ndufreche** [comment]
:+1:

**Pull request merged**